Task 4/Task4.1/Task4.1.1/DirectoryOverseer.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.IO;
+using Newtonsoft.Json;
+
+namespace Task4._1._1
+{
+    /// <summary>
+    /// Class to watch over txt files and logs their changes
+    /// </summary>
+    public class DirectoryOverseer
+    {
+        /// <summary>
+        /// The file to log to and to get old logs
+        /// </summary>
+        readonly string _changeLogPath;
+
+        FileSystemWatcher _fileSystemWatcher;
+
+        /// <summary>
+        /// All file change events, except renamed
+        /// </summary>
+        public event FileSystemEventHandler FileAltered;
+
+        public event RenamedEventHandler FileRenamed;
+
+        public DirectoryOverseer(string directoryPath, 
+            string changeLogPath = @"changes\log.json") 
+        {
+            DirectoryPath = directoryPath;
+
+            _changeLogPath = changeLogPath;
+
+            _fileSystemWatcher =
+                new FileSystemWatcher(DirectoryPath, "*.txt");
+
+            DeserializeLog();
+
+            _fileSystemWatcher.Changed += OnChange;
+            _fileSystemWatcher.Created += OnCreate;
+            _fileSystemWatcher.Deleted += OnDelete;
+            _fileSystemWatcher.Renamed += OnRename;
+
+            _fileSystemWatcher.IncludeSubdirectories = true;
+            _fileSystemWatcher.EnableRaisingEvents = true;
+        }
+
+        /// <summary>
+        /// The path of the Directory to monitor
+        /// </summary>
+        public string DirectoryPath
+        {
+            get;
+
+            private set;
+        }
```

Comment:
```suggestion
        public string DirectoryPath { get; private set; }
```
Не надо разворачивать свойства в несколько строк, если они не содержат дополнительной логики в гетерах и сеттерах